### PR TITLE
Add support for GitHub Actions

### PR DESCRIPTION
**Pull request by doomspork**
Howdy @rvirding 👋 First and foremost, thanks for everything you do 🙏 We talked briefly in Slack about luerl and since those conversations I've been leveraging this library to empower a number of production systems so it's high time I start giving back. 

I was looking through the issues and PRs on the repo and noticed #89 which is now many years old. With the availability of GitHub Actions I think it might make more sense to go that route than TravisCI — are you open to that?  This PR introduces a single action for running tests. Happy to expand on this with your guidance.

One call out I did want to make: It seems there is a failing test on `develop` currently but when this change is ported to the `master` branch we can verify this action runs and tests all pass (here's an example PR on my fork: https://github.com/doomspork/luerl/pull/2).

Please let me know if there's any additional changes you'd like to make. Happy to incorporate any changes that would make this more useful to yourself and other contributors.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - synchronize
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: Erlang/OTP ${{matrix.otp}}
+    strategy:
+      matrix:
+        otp: ['24.3', '25.3']
+        rebar3: ['3.20']
```
Comment (doomspork) [inline .github/workflows/ci.yml:18]: Should we expand the matrix to cover a greater spectrum of OTP and rebar3 versions?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          rebar3-version: ${{matrix.rebar3}}
+      - run: rebar3 do eunit, ct --cover, cover
```
Comment (doomspork) [inline .github/workflows/ci.yml:25]: Are there any additional commands we should consider running? Separately, would you prefer `eunit` and `cover` be pulled into separate steps?

Comment (doomspork) [reply .github/workflows/ci.yml:25]: An example draft PR on my fork with the steps broken out: https://github.com/doomspork/luerl/pull/3

image

**Comment by doomspork** [comment]
Thanks for the merge @rvirding! 🎉

**Comment by rvirding** [comment]
A bit late but I think we should expand the matrix to cover more versions of OTP and rebar3. This I will do.

**Comment by doomspork** [comment]
@rvirding happy to help! If you want to let me know your thoughts on the matrix I'll get a PR up

**Pull request merged**